src/displayapp/screens/Twos.cpp

```diff
@@ -97,39 +92,37 @@ Twos::~Twos() {
 }
 
 bool Twos::placeNewTile() {
-  std::vector<std::pair<int, int>> availableCells;
-  for (int row = 0; row < 4; row++) {
-    for (int col = 0; col < 4; col++) {
-      if (!grid[row][col].value) {
-        availableCells.push_back(std::make_pair(row, col));
-      }
+  unsigned int emptyCells[nCells];
+  unsigned int nEmpty = 0;
+  for (unsigned int i = 0; i < nCells; i++) {
+    const unsigned int row = i / nCols;
+    const unsigned int col = i % nCols;
+    if (grid[row][col].value == 0) {
+      emptyCells[nEmpty] = i;
+      nEmpty++;
```

Comment:
suggest using nested loop and less costly multiplications instead of costly division and modulo operations
```suggestion
  for (unsigned int row = 0; row < nRows; row++) {
    for (unsigned int col = 0; col < nCols; col++) {
      if (grid[row][col].value == 0) {
        emptyCells[nEmpty] = row * nCols + cols;
        nEmpty++;
      }
```

Reply:
Division and modulo get optimized to bitshift and `AND`. The current solution is a few bytes smaller than the suggestion.

Reply:
In this case, the modulo operations shouldn't be costly because nCols is a power of 2. Modulo operations with powers of 2 should be optimised by the compiler to an AND operation with (in this case) 3.

Reply:
Actually it simply treats the array as one dimensional and eliminates the variables.